### PR TITLE
[Notifier] deprecate the TransportFactoryTestCase

### LINKED ISSUE #54100: [Notifier] TransportFactoryTestCase incompatible with PHPUnit 11
### Symfony version(s) affected

7.1-dev

### Description

`TransportFactoryTestCase` contains several data providers that return an empty array by default, for example:

https://github.com/symfony/symfony/blob/9e810dea50cb47deb8500ad7ffb56d5a5622b46e/src/Symfony/Component/Notifier/Test/TransportFactoryTestCase.php#L41-L47

If I run such a test with PHPUnit 11, the test in marked as invalid if the data provider is left empty.

### How to reproduce

```
There was 1 PHPUnit error:

1) Symfony\Component\Notifier\Bridge\Sms77\Tests\Sms77TransportFactoryTest::testMissingRequiredOptionException
The data provider specified for Symfony\Component\Notifier\Bridge\Sms77\Tests\Sms77TransportFactoryTest::testMissingRequiredOptionException is invalid
Empty data set provided by data provider
```

### Possible Solution

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 7.2
| Bug fix?      | no
| New feature?  | no
| Deprecations? | yes
| Issues        | Fix #54100
| License       | MIT

### CODE REVIEW

**Comment by derrabus** [comment]
Does this really solve this problem? Implementers must not implement those methods by returning an empty array either. And if not providing any test cases is legitimate, what should the implementer do with those methods?

**Comment by OskarStark** [comment]
> And if not providing any test cases is legitimate, what should the implementer do with those methods?

I agree, back in time I raised this question to Sebastian, but with no luck. It makes things way more complicated than they have to be 🤷

**Comment by xabbuh** [comment]
The new `AbstractTransportFactoryTestCase` defines just three tests (`testSupports()`, `testCreate()`, and `testUnsupportedSchemeException()`) for all which reasonable test data can be drafted for a transport factory.

There were two tests (`testIncompleteDsnException()` and `MissingRequiredOptionTestTrait`) that do not make sense for all transport factories. Those have been moved into traits.

File: UPGRADE-7.2.md [hunk 1]
```diff
@@ -52,6 +52,14 @@ Messenger
 
  * Add `getRetryDelay()` method to `RecoverableExceptionInterface`
 
+Notifier
+--------
+
+ * Deprecate `TransportFactoryTestCase`, implement `AbstractTransportFactoryTestCase` instead
```
Comment (stof) [inline UPGRADE-7.2.md]:
```suggestion
 * Deprecate `TransportFactoryTestCase`, extend `AbstractTransportFactoryTestCase` instead
```

**Review by stof — APPROVED** [review]

**Review by stof — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**